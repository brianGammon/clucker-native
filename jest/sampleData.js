const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
