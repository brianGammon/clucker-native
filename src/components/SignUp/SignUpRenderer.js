import * as React from 'react';
import { FieldGroup, FieldControl } from 'react-reactive-form';
import { Button, Text, Form } from 'native-base';
import FormInput from '../FormInput';
import AuthContainer from '../Auth/AuthContainer';
import AuthExtraLink from '../Auth/AuthExtraLink';

type Props = {
  navigation: any,
  error: string,
  loginForm: any,
  handleSubmit: any,
};

const SignUpRenderer = ({
  navigation,
  error,
  loginForm,
  handleSubmit,
}: Props) => (
  <AuthContainer
    title="Sign Up"
    error={error}
    extraLinks={[
      <AuthExtraLink
        key={0}
        message="Already have an account?"
        buttonText="Sign In"
        handlePress={() => navigation.replace('SignIn')}
      />,
    ]}
  >
    <FieldGroup
      control={loginForm}
      render={({ invalid }) => (
        <Form>
          <FieldControl
            name="email"
            render={formProps => <FormInput {...formProps} />}
            meta={{
              label: 'Email',
              autoCapitalize: 'none',
              keyboardType: 'email-address',
            }}
          />

          <FieldControl
            name="password"
            render={formProps => <FormInput {...formProps} />}
            meta={{
              label: 'Password',
              autoCapitalize: 'none',
              secureTextEntry: true,
            }}
          />

          <Button
            style={{ marginTop: 10 }}
            block
            onPress={handleSubmit}
            disabled={invalid}
          >
            <Text>Submit</Text>
          </Button>
        </Form>
      )}
    />
  </AuthContainer>
);

export default SignUpRenderer;
