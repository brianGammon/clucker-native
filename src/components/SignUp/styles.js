import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  formContainer: {
    width: '100%',
    alignItems: 'center',
  },
  error: {
    color: 'red',
    padding: 20,
  },
  signUpContainer: {
    backgroundColor: 'rgba(255, 255, 255, .85)',
    width: '90%',
    padding: 20,
    borderRadius: 20,
  },
  title: {
    fontSize: 40,
    fontWeight: 'bold',
    alignSelf: 'center',
  },
});
