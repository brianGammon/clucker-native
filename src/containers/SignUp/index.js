import * as React from 'react';
import { Text, View, Button } from 'react-native';

type Props = {
  navigation: any,
};

export default class SignUp extends React.Component<Props> {
  static navigationOptions = {
    title: 'Sign Up',
  };

  render() {
    const { navigation } = this.props;
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text>Sign Up Screen</Text>
        <Button onPress={() => navigation.replace('SignIn')} title="Sign In" />
        <Button onPress={() => navigation.navigate('ResetPassword')} title="Reset Password" />
      </View>
    );
  }
}
