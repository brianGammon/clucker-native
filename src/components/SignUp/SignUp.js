/* @flow */
import * as React from 'react';
import { connect } from 'react-redux';
import { FormBuilder, Validators } from 'react-reactive-form';
import SignUpRenderer from './SignUpRenderer';
import { signUpRequested } from '../../redux/actions';
import { actionTypes } from '../../redux/constants';
import { mustMatchValidator } from '../../utils/validators';

type Props = {
  navigation: any,
  error: string,
  signUp: (email: string, password: string) => void,
  clearError: () => void,
};

class SignUp extends React.Component<Props> {
  static navigationOptions = {
    title: 'Sign In',
  };

  loginForm = FormBuilder.group(
    {
      email: ['', [Validators.email, Validators.required]],
      password: ['', [Validators.required, Validators.minLength(6)]],
      confirmPassword: ['', Validators.required],
    },
    { validators: mustMatchValidator('confirmPassword', 'password') },
  );

  componentWillUnmount() {
    const { error, clearError } = this.props;
    if (error) {
      clearError();
    }
  }

  handleSubmit = () => {
    const { email, password } = this.loginForm.value;
    const { signUp } = this.props;
    signUp(email, password);
  };

  render() {
    const { navigation, error } = this.props;
    return (
      <SignUpRenderer
        navigation={navigation}
        error={error}
        loginForm={this.loginForm}
        handleSubmit={this.handleSubmit}
      />
    );
  }
}

const mapStateToProps = ({ auth: { errors } }) => ({
  error: errors.signUp,
});

const mapDispatchToProps = dispatch => ({
  signUp: (email, password) => dispatch(signUpRequested(email, password)),
  clearError: () => dispatch({ type: actionTypes.CLEAR_AUTH_ERROR }),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SignUp);
