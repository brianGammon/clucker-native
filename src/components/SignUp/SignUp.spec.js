import React from 'react';
import configureStore from 'redux-mock-store';
import sampleData from '../../../jest/test-data/no-flocks.json';
import SignUp from '.';
import SignUpRenderer from './SignUpRenderer';

describe('SignUp component:', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore(sampleData);
  test('Should render', () => {
    const wrapper = shallow(<SignUp store={store} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test('SignUpRenderer renders when no error message', () => {
    const wrapper = shallow(
      <SignUpRenderer
        navigation={() => {}}
        email=""
        password=""
        error=""
        handleSignUp={() => {}}
        handleChangeText={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('SignUpRenderer renders with error message', () => {
    const wrapper = shallow(
      <SignUpRenderer
        navigation={() => {}}
        email=""
        password=""
        error="Some error occurred"
        handleSignUp={() => {}}
        handleChangeText={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
