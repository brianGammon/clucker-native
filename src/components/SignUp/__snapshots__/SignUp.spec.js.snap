// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component: Should render 1`] = `
<SignUpRenderer
  error={null}
  focusNext={[Function]}
  handleSubmit={[Function]}
  inputRefs={Object {}}
  loginForm={
    FormGroup {
      "_onDisabledChange": Array [],
      "_pendingChange": false,
      "_pendingDirty": false,
      "_pendingTouched": false,
      "asyncValidator": null,
      "controls": Object {
        "email": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "",
          "active": false,
          "asyncValidator": null,
          "errors": Object {
            "required": true,
          },
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "INVALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": [Function],
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": Array [
              [Function],
              [Function],
            ],
          },
          "value": "",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "password": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "",
          "active": false,
          "asyncValidator": null,
          "errors": Object {
            "required": true,
          },
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "INVALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": [Function],
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": Array [
              [Function],
              [Function],
            ],
          },
          "value": "",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
      },
      "errors": null,
      "get": [Function],
      "getError": [Function],
      "handleSubmit": [Function],
      "hasError": [Function],
      "meta": Object {},
      "patchValue": [Function],
      "pristine": true,
      "reset": [Function],
      "setValue": [Function],
      "stateChanges": Observable {
        "observers": Array [],
      },
      "status": "INVALID",
      "statusChanges": Observable {
        "observers": Array [],
      },
      "submitted": false,
      "touched": false,
      "validator": null,
      "validatorOrOpts": Object {
        "asyncValidators": null,
        "updateOn": null,
        "validators": null,
      },
      "value": Object {
        "email": "",
        "password": "",
      },
      "valueChanges": Observable {
        "observers": Array [],
      },
    }
  }
/>
`;

exports[`SignUp component: SignUpRenderer renders when no error message 1`] = `
<AuthBase
  error=""
  extraLinks={
    Array [
      <AuthExtraLink
        buttonText="Sign In"
        handlePress={[Function]}
        message="Already have an account?"
      />,
    ]
  }
  title="Sign Up"
>
  <FieldGroup
    render={[Function]}
    strict={true}
  />
</AuthBase>
`;

exports[`SignUp component: SignUpRenderer renders with error message 1`] = `
<AuthBase
  error="Some error occurred"
  extraLinks={
    Array [
      <AuthExtraLink
        buttonText="Sign In"
        handlePress={[Function]}
        message="Already have an account?"
      />,
    ]
  }
  title="Sign Up"
>
  <FieldGroup
    render={[Function]}
    strict={true}
  />
</AuthBase>
`;
